### ISSUE TITLE
Dropout between convolution and bias_add

### ISSUE THREAD

**Issue by yiqianglee** [jakeret/tf_unet#187]
Is there any reason to put dropout between convolution and bias_add? 
Generally speaking, we should do dropout after adding bias, because bias_add is part of convolution, right?

/////////////////////////////////////////////////////////
        conv1 = conv2d(in_node, w1, keep_prob)  
        tmp_h_conv = tf.nn.relu(conv1 + b1)

**Comment by yiqianglee** [comment]
any comments?

**Comment by wkeithvan** [comment]
Yeah... you're correct. That's definitely an error. The original paper (https://arxiv.org/abs/1207.0580) and all the literature I could find talks about removing the output of a unit from the network. With bias still left in, that would not remove the output, only the input*weight part. This would lead to incorrect training of the bias element.

**Comment by wkeithvan** [comment]
I've added a pull request to fix this

**Comment by yiqianglee** [comment]
Thanks for your quick action. @wkeithvan 
Do you think it's reasonable to use tf.nn.bias_add, not directly use "+"? 
I think they are exact same semantic in your case, but will make code much better readability and performance friendly (backend  can much more easily to do optimization for bias_add op)? Thanks!

**Comment by wkeithvan** [comment]
Interesting... I'm relatively new to ML and TensorFlow, so I hadn't seen this function. I'll definitely take a look and see if it is implementable. We should move the implementation conversation over to the pull request and continue there!
